File: src/main/java/discordPanel/DiscordPanelFactory.java
package discordPanel;

import com.intellij.openapi.project.DumbAware;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowFactory;
import com.intellij.ui.content.Content;
import com.intellij.ui.content.ContentFactory;
import org.jetbrains.annotations.NotNull;

public class DiscordPanelFactory  implements ToolWindowFactory, DumbAware {
    // Create the tool window content.
    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
        DiscordPanelImpl myToolWindow = new DiscordPanelImpl(toolWindow);
        ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
        Content content = contentFactory.createContent(myToolWindow.getContent(), "Discord Panel", false);
        toolWindow.getContentManager().addContent(content);
    }
}

File: src/main/kotlin/discordPanel/DiscordPanelFactory.kt
package discordPanel

import com.intellij.openapi.project.DumbAware
import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.openapi.wm.ToolWindowFactory
import com.intellij.ui.content.ContentFactory

class DiscordPanelFactory : ToolWindowFactory, DumbAware {
    // Create the tool window content.
    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
        val myToolWindow = DiscordPanelImpl(toolWindow)
        val contentFactory = ContentFactory.SERVICE.getInstance()
        val content = contentFactory.createContent(myToolWindow.content, "Discord Panel", false)
        toolWindow.contentManager.addContent(content)
    }
}